staging/src/k8s.io/apiserver/pkg/admission/update.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// fieldManagerAdmissionWrapper is an instance of admission.NamedHandler that wraps other admission handlers for keeping track of their changes and updating their managedFields accordingly
+type fieldManagerAdmissionWrapper struct {
+	Interface
+
+	manager string
+}
+
+// NewFieldManagerAdmissionWrapper creates a new handler wrapping the passed in handler
+func NewFieldManagerAdmissionWrapper(handler Interface, manager string) Interface {
```

Comment:
I don't see where this is called, what am I missing? 

Also there is something that seems weird from a lifespan perspective. My understanding is that the lifespan goes like this: `admit.Interface > FieldManager > manager`, so why would manager be in the class but `FieldManager` passed to each individual call?

Reply:
It's being called here: https://github.com/kubernetes/kubernetes/pull/78469/files#diff-cb72549ce8ab8dfb0c25497e2580d3b1R143
The reason it is this way is, that the manager name is the plug-in name, which is not available at the request level. On the other hand the FieldManager is only available there.

Did you see my comment above about how it might be cleaner adding GetObjectFieldManager() to admission.ObjectInterfaces instead?

Reply:
Why is the field manager only available at a request level? We create that and give it to the handlers when they are installed.

Would it make sense to instead wrap the admission thing defined here: https://github.com/kubernetes/kubernetes/blob/0184c6c4a7c3eb8e300b1552d5957c9fb4387f4a/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L184 right after the fieldmanager is created?

Reply:
Yes, wrapping this would be an option. The downside is, we would wrap the entire admission chain so we can not know which controller changed which field but the field owner would be something like "AdmissionControllers".
We might have to do that anyways as I don't know how running an update for every controller impacts performance. What do you think about it?

Reply:
We could also think about ways to make name information available for every plugin and then put the updating into the chain handler, but that would involve a lot of change as well.